Templates/AppSource App/.github/workflows/CICD.yaml

```diff
@@ -207,7 +214,7 @@ jobs:
 
   Deploy:
     needs: [ Initialization, Build ]
-    if: always() && needs.Build.result == 'Success' && needs.Initialization.outputs.environmentCount > 0
+    if: (!cancelled()) && (needs.Build.result == 'success' || needs.Build.result == 'skipped') && needs.Initialization.outputs.environmentCount > 0
```

Comment:
If all build jobs are skipped, there'll be no artifacts to deploy. Is this case handed by the `Deploy` job?

As far as I remember, CICD does not run incremental build, it always builds all projects, so the `needs.Build.result == 'skipped'` check seems unnecessary.

Reply:
If you remove a dependency, the dependency depth might change and this will still work without needing to update AL-Go System Files.